scanners/seo.py

```diff
@@ -56,46 +81,68 @@ def init(environment: dict, options: dict) -> dict:
+        sitemap_status = "OK"
+    else:
+        sitemap_status = sitemap_results['status_code']
+
     results = {
-        'robots': requests.get("https://" + domain + '/robots.txt').status_code,
-        'warnings': {}
+        'Platforms': 'Unknown',
+        'Sitemap.xml': sitemap_status,
+        'Sitemap Final URL': sitemap_results['final_url'],
+        'Sitemap items': sitemap_results['url_tag_count'],
+        'PDFs in sitemap': sitemap_results['pdfs_in_urls'],
+        'Sitemaps from index': sitemap_results['sitemap_locations_from_index'],
+        'Robots.txt': sitemap_results['robots'],
+        'Crawl delay': sitemap_results['crawl_delay'],
+        'Sitemaps from robots': sitemap_results['sitemap_locations_from_robotstxt'],
+        'Total URLs': sitemap_results['url_tag_count'] if sitemap_results['url_tag_count'] else 0,
+        'Est time to index': 'Unknown',
+        'Main tags found': False,
+        'Search found': False,
+        'Warnings': {},
     }
-    # Now let's take a look at sitemap
-    sitemap = requests.get("https://" + domain + '/sitemap.xml')
-    sitemap_status = sitemap.status_code
-    results['sitemap'] = {
-        'status': sitemap_status
-    }
-    # If we have found a sitemap, see how many URLs are in there?
-    # TO DO: If sitemap.xml redirects to a sitemap index,
-    # we're not handling that correctly yet. We need to detect that
-    # redirect, then ... what? Follow from there?
-    if sitemap_status == HTTPStatus.OK:
-        soup = BeautifulSoup(sitemap.text, 'xml')
-        urls = soup.find_all('url')
-        results['sitemap']['urls found'] = len(urls)
-        # and how many of those URLs appear to be PDFs
-        if urls:
-            pdfcount = len([u for u in urls if '.pdf' in u.get_text()])
-            results['sitemap']['PDFs found in sitemap'] = pdfcount
+
+    # See if we can determine platforms used for the site
+    build_info = builtwith(fqd)
+    if 'web-frameworks' in build_info:
+        results['Platforms'] = build_info['web-frameworks']
+
+    # If we found additional sitemaps in a sitemap index or in robots.txt, we
+    # need to go look at them and update our url total.
+    additional_urls = 0
+    for loc in sitemap_results['sitemap_locations_from_index']:
+        if loc != sitemap_results['final_url']:
+            sitemap = requests.get(loc)
+            if sitemap.status_code == HTTPStatus.OK:
+                soup = BeautifulSoup(sitemap.text, 'xml')
+                additional_urls += len(soup.find_all('url'))
+
+    for loc in sitemap_results['sitemap_locations_from_robotstxt']:
+        if loc != sitemap_results['final_url']:
+            sitemap = requests.get(loc)
+            if sitemap.status_code == HTTPStatus.OK:
+                soup = BeautifulSoup(sitemap.text, 'xml')
+                additional_urls += len(soup.find_all('url'))
+    results['Total URLs'] = results['Total URLs'] + additional_urls
```

Comment:
The desire (at least for Search) is to get a total of all the URLs we could find and will need to index, which we'll then multiply by the crawl rate (if we got it) to get a sense of how long it would take to index the whole enchilada. I considered checking for dupes but honestly didn't figure it was worth the overhead given that non-exact is OK.